sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -2104,7 +2104,7 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
 
   def ambiguousReferenceToFieldsError(fields: String): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1209",
+      errorClass = "AMBIGUOUS_REFERENCE_TO_FIELDS",
       messageParameters = Map("fields" -> fields))
```

Comment:
Fields should be quoted by `toSQLId()`

Reply:
Done. Also I've changed the error message for this, please check. (added number of field appearance in the schema)